kong/plugins/mashape-analytics/buffer.lua

```diff
@@ -26,15 +32,27 @@ local ngx_timer_at = ngx.timer.at
 local table_insert = table.insert
 local table_concat = table.concat
 local table_remove = table.remove
+local string_sub = string.sub
 local string_len = string.len
 local string_rep = string.rep
 local string_format = string.format
+local math_pow = math.pow
+local math_min = math.min
 local setmetatable = setmetatable
 
 local MB = 1024 * 1024
-local MAX_BUFFER_SIZE = 1 * MB
+local MAX_BUFFER_SIZE = 500 * MB
```

Comment:
This value is correct. This constant will be renamed, and its actual purpose is to mirror a constant present in the Galileo collector. It is the maximum payload size accepted by an HTTP request by the Galileo collector. It used to be 1MB, and is now 500. I believe this value should mirror the Galileo restrictions in the case someone really knows what he is doing and wish to send extremely large batches of ALFs (possibly containing request bodies that are very large). This value is hard coded and cannot be bypassed. Any ALF beyond that limit will be dropped as it is too big by itself, and if a batch reaches that limit, it will be put in the sending_queue directly, wether it has reached the configured `batch_size` or not.

`batch_size` (n of ALFs) and `sending_queue_size` (Mbs) are the value one should tweak to enforce a low memory footprint. I could see a third value, being the `batch_size` but in terms of bytes too, being useful. In case one's ALFs are much much bigger than he intended to. It would make the plugin more complicated to understand and configure (it already does with the `sending_queue_size` so I am not sure about that yet.